lib/api-tests/tests/high_level_api.rs

```diff
@@ -0,0 +1,177 @@
+static TEST_WAT: &str = r#"
+(module
+  (import "env" "outside-global" (global $outside-global (mut i32)))
+  (import "env" "update-func" (func $update-func (param i32) (result i32)))
+  (table $test-table (export "test-table") 2 anyfunc)
+  (global $test-global (export "test-global") (mut i32) (i32.const 3))
+  (elem (;0;) (i32.const 0) $ret_2)
+  (func $ret_2 (export "ret_2") (result i32)
+    i32.const 2)
+  (func $ret_4 (export "ret_4") (result i32)
+    i32.const 4)
+  (func $set_test_global (export "set_test_global") (param i32)
+    (global.set $test-global
+                (local.get 0)))
+  (func $update-outside-global (export "update_outside_global")
+    (global.set $outside-global
+                (call $update-func (global.get $outside-global))))
+)
+"#;
+
+fn append_custom_section(
+    wasm: &[u8],
+    custom_section_name: &str,
+    custom_section_contents: &[u8],
+) -> Vec<u8> {
+    let mut out = Vec::with_capacity(
+        // 1 for custom section id, 5 for max length of custom section, 5 for max length of name
+        wasm.len() + custom_section_name.len() + custom_section_contents.len() + 1 + 5 + 5,
+    );
+
+    out.extend(wasm);
+    out.push(0);
+
+    let name_length = custom_section_name.as_bytes().len() as u32;
+    let mut name_length_as_leb128 = vec![];
+    write_u32_leb128(&mut name_length_as_leb128, name_length);
+
+    let custom_section_length = (custom_section_contents.len()
+        + custom_section_name.as_bytes().len()
+        + name_length_as_leb128.len()) as u32;
+
+    let mut custom_section_length_as_leb128 = vec![];
+    write_u32_leb128(&mut custom_section_length_as_leb128, custom_section_length);
+
+    out.extend(&custom_section_length_as_leb128);
+    out.extend(&name_length_as_leb128);
+    out.extend(custom_section_name.as_bytes());
+    out.extend(custom_section_contents);
+
+    out
+}
+
+#[test]
+fn it_works() {
```

Comment:
Right now we are testing a lot of different things in one, which makes harder to spot multiple issues as one function will fail in cascade.

We are testing:
1. Custom sections
2. Module exports (should be ordered)
3. Instantiation
4. Variables working without instantiating them:
   * Outside global
4. Exports working
   * Functions: function retrieval and function calls
   * Tables: table retrieval
   * Globals: globals retrieval and globals getters

I think all those should be tested separately